From 84259714c2376fdb919299e8e01af1ccc6fbe2c0 Mon Sep 17 00:00:00 2001
From: Stefan Schmidt <s.schmidt@samsung.com>
Date: Thu, 12 Feb 2015 15:24:27 +0100
Subject: [PATCH 3/3] edje_cc: Avoid segfault when strcmp NULL items

Martin reported a porblem with one of the SHR project edc files after an upgrade
to 1.13. The segfault itself is easy enough to fix here but I think there might
be other problems with the edc after this segfault is gone. We will see.

Fixes T2106
---
 src/bin/edje/edje_cc_out.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/bin/edje/edje_cc_out.c b/src/bin/edje/edje_cc_out.c
index 4f582c7..bc85444 100755
--- a/src/bin/edje/edje_cc_out.c
+++ b/src/bin/edje/edje_cc_out.c
@@ -386,7 +386,7 @@ check_source_links(Edje_Part_Collection *pc, Edje_Part *ep, Eet_File *ef, Eina_L
    EINA_LIST_FOREACH(edje_collections, l, pc_source)
      {
         /* Find sourced group */
-        if (strcmp(ep->source, pc_source->part) == 0)
+        if (ep->source && pc_source->part && strcmp(ep->source, pc_source->part) == 0)
           {
              /* Go through every part to find parts with type GROUP */
              for (i = 0; i < pc_source->parts_count; ++i)
-- 
2.3.0

